File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    
    int count = 0;
    int zero = 0;
    
    while(s != "1"){
        count++;
        
        string temp = "";
        
        for(int i=0;i<s.size();i++){
            if(s[i] == '1'){
                temp += "1";
            }
            else if(s[i] == '0'){
                zero++;
            }
        }
        
        s = "";
        int num = temp.size();
        
        while(num){
            s += to_string(num % 2);
            num /= 2;
        }
    }
    
    answer.push_back(count);
    answer.push_back(zero);
    
    return answer;
}

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.cs
using System;
using System.Collections.Generic;

public class Solution {    
    public string RemoveX(string str, out int count){
        string temp = "";
        count = 0;
        
        for(int i=0;i<str.Length;i++){
            if(str[i] == '1'){
                temp += "1";
            }
            else{
                count++;
            }
        }
        return temp;
    }
    
    public string LengthToBinary(string str){
        int len = str.Length;
        return Convert.ToString(len,2);
    }
    
    public int[] solution(string s) {
        int changeCount = 0;
        int zeroCount = 0;
        
        while(s != "1"){
            int count;
            s = RemoveX(s, out count);
            zeroCount += count;

            s = LengthToBinary(s); 
            changeCount++;
        }
        
        return new int[] { changeCount, zeroCount };
    }
}